qualang_tools/config/components.py

```diff
@@ -684,14 +736,36 @@ def smearing(self, smearing: int):
         self.dict["smearing"] = smearing
 
 
+class PiecewiseConstantIntegrationWeights(IntegrationWeights):
+    def __init__(self, name: str, cosines: List, sines: List, durations: List):
+        """Piece-wise constant integration weights used in integration and demodulation
+        :param name: name for the weights vector
+        :type name: str
+        :param cosines: values of the cosine vector
+        :type cosines: List[float]
+        :param sines: values of the sine vector
+        :type sines: List[float]
+        :param durations: duration of the each segment
+        :type durations: List[int]
+        """
+        self._durations = durations
+        self._cosines = cosines
+        self._sines = sines
+        super(PiecewiseConstantIntegrationWeights, self).__init__(
+            name,
+            [(cosine, duration) for (cosine, duration) in zip(cosines, durations)],
```

Comment:
The durations of the cosine weights segments doesn't have to match the duration of the sine weights segments

Reply:
Are you sure?

Reply:
Ummmm nope... I'll check

Reply:
Actually, I think they have to otherwise: `qm - ERROR - CONFIG ERROR in key "integrationWeights.Wc" [Integration weights] : Integration weight Wc sine and cosine size is not equal`

Reply:
So I meant that the piecewise segments don't have to match in duration (or number). Of course, the total length must be the same. 

For example, the following is valid:
```
"cosine_weights": {
    "cosine": [(1.0, readout_len/2),(0.1, readout_len/2)],
    "sine": [(0.0, readout_len)],
},
```